src/python/pants/util/dirutil.py

```diff
@@ -66,7 +67,15 @@ def safe_mkdir(directory, clean=False):
   if clean:
     safe_rmtree(directory)
   try:
-    os.makedirs(directory)
+    if directory.endswith(get_repo_root() + '/.pants.d'):
```

Comment:
It'd be better to use pathlib.PurePath(directory).name == ".pants.d" instead of checking if the path string ends with "/.pants.d", which will fail if the path ends with "/.pants.d/" for instance.

Reply:
Can't use `== ".pants.d"` because there  are sub-dirs with the same name (".pants.d"). But added `rstrip('/')`  to ensure the directory path doesn't have `/` prefix. E.g. : `if directory.rstrip('/').endswith(get_workdir_symlink())`